src/AdoPat/PatClient.cs

```diff
@@ -0,0 +1,139 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+namespace Microsoft.Authentication.AdoPat
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using Microsoft.VisualStudio.Services.DelegatedAuthorization;
+
+    /// <summary>
+    /// An abstraction over common operations with the Azure DevOps PAT Lifecycle Management REST API.
+    /// </summary>
+    public class PatClient
+    {
+        private const int PageSize = 100; // Using the maximum allowable page size allows us to reduce HTTP calls.
+        private const bool AllOrgs = false; // Azure DevOps have recommended we always set this to false.

```

Comment:
[nit] I couldn't make out from here what `AllOrgs` is. And may be provide a reference to it about setting it to false.

Reply:
@mvanchaa: Does [this documentation](https://learn.microsoft.com/en-us/rest/api/azure/devops/tokens/pats/create?view=azure-devops-rest-7.1&tabs=HTTP#pattokencreaterequest) help make it more clear what `allOrgs` is?

Reply:
yes :)

Reply:
Awesome, I'll include a link or copy/paste that text here to better explain what `allOrgs` is about. 😁

Reply:
The documentation link is pretty long, so I just described what the field does in a comment.